s3.go

```diff
@@ -55,6 +56,93 @@ func initS3() {
 	})
 }
 
+func transparentS3Head(c *gin.Context) {
+	bucket := c.Param("bucket")
+	key := c.Param("key")
+
+	res, err := s3Client.HeadObject(&s3.HeadObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(key),
+	})
+	if err != nil {
+		msg := fmt.Sprintf("Failed to get header for '%s#%s' from S3: %v", bucket, key, err)
+		log.Errorf(msg)
+		c.JSON(500, gin.H{"error": msg})
```

Comment:
we're a bit inconsistent as to what error codes we return in the other transparent APIs. Ideally we'd pass through the error code but `S3Client` seems to have swallowed it...

Reply:
![image](https://user-images.githubusercontent.com/237927/155348604-4c904907-b2e0-4137-a2cd-e6987faf86a6.png)
![image](https://user-images.githubusercontent.com/237927/155349131-9cb9a672-949c-4981-8a2e-f36d772cb19a.png)

I see these as potential errors returned. Also I'd expect `err` to actually contain at least the 404/403 error which we can forward? 

Either way defaulting to 500 would not be a good idea here unfortunately.